packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -54,6 +54,11 @@ export interface AriaSelectableCollectionOptions {
    * @default false
    */
   disallowSelectAll?: boolean,
+  /**
+   * Whether the collection allows the user to clear all selected items via Escape.
+   * @default false
+   */
+  disallowClearAll?: boolean,
```

Comment:
Open question: I've added this prop to ListBox and Menu, but do we want to propagate it to the rest of the selectable collections? Opted for the smaller change for now (aka only covering the Autocomplete case), but happy to cover all selectable collections where it makes (aka GridList/Table but not Picker/ComboBox/Tabs).

Reply:
I think it should be in the other collections as well. The use case is already being built for gridlist

https://github.com/adobe/react-spectrum/issues/7105
https://github.com/adobe/react-spectrum/pull/7922

Reply:
sounds good, I'll add it then